metalearners/_narwhals_utils.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) QuantCo 2024-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+from collections.abc import Sequence
+from types import ModuleType
+
+import narwhals.stable.v1 as nw
+import numpy as np
+from narwhals.dependencies import is_into_series
+
+from metalearners._typing import Vector
+
+
+def nw_to_dummies(
+    x: nw.Series, categories: Sequence, column_name: str, drop_first: bool = True
+) -> nw.DataFrame:
+    """Turn a vector into a matrix with dummies.
+
+    This operation is also referred to as one-hot-encoding.
+
+    ``x`` is expected to have values which can be cast to integer.
+    """
+    if len(categories) < 2:
+        raise ValueError(
+            "categories to be used for nw_to_dummies must have at least two "
+            "distinct values."
+        )
+
+    if set(categories) < set(x.unique()):
+        raise ValueError("We observed a value which isn't part of the categories.")
+
+    relevant_categories = categories[1:] if drop_first else categories
+    return x.to_frame().select(
+        [
+            (nw.col(column_name) == cat).cast(nw.Int8).name.suffix(f"_{cat}")
```

Comment:
nice! just make sure / test that null handling works as expected – whatever your definition of that is: Do we want nulls to turn into a row of nulls in the OHE matrix or should it be a row of zeros?

Reply:
Excellent point!

f535c40